trolley/recipient_gateway.py

```diff
@@ -57,12 +58,45 @@ def delete(self, recipient_id):
             self.config).delete(endpoint)
         return True
 
-    def search(self, page=None, page_size=None, search=None, name=None, email=None, reference_id=None, start_date=None,
+    """ Search Recipients with a search term.
+        This method returns a generator which auto paginates.
+        You can use this generator in a foreach loop to sequentially go through all the 
+        search results without needing to call this method again.
+        
+            For accessing specific pages, check the search_by_page() method. """
+    def search(self, search=None, name=None, email=None, reference_id=None, start_date=None,
                end_date=None, status=None, compliance_status=None, country=None, payout_method=None, currency=None,
                order_by=None, sort_by=None):
+        
+        local_vars = UrlUtils.parse(locals())
+        
+        page = 0
+        should_paginate = True
+        while should_paginate:
+            page+=1
+            endpoint = f'/v1/recipients?page={page}&pageSize=10' + (f'&{local_vars}' if(len(local_vars)) else '')
+            response = trolley.configuration.Configuration.client(self.config).get(endpoint)
+            yield from self.__build_recipients_from_response(response, False)
+
+            if page < response["meta"]["pages"]:
+                should_paginate = True
+            else:
+                should_paginate = False
+        
+        return self.__build_recipients_from_response(response)
```

Comment:
In line 78, already yield all the result, why do we still need to return it here?

Reply:
Good catch.
I'll fix this